packages/ubuntu_desktop_installer/lib/services/telemetry_service.dart

```diff
@@ -0,0 +1,67 @@
+import 'dart:convert';
+
+import 'package:file/file.dart';
+import 'package:file/local.dart';
+import 'package:flutter/widgets.dart';
+import 'package:system_clock/system_clock.dart';
+
+class TelemetryService {
+  TelemetryService() {
+    _startTime = _uptime();
+  }
+
+  static const reportLocation = '/target/var/log/installer/telemetry';
+
+  final Map<String, dynamic> _metrics = {};
+  final Map<String, String> _stages = {};
+  bool _done = false;
+  late int _startTime;
+
+  int _uptime() {
+    try {
+      return SystemClock.uptime().inSeconds;
+      // ignore: avoid_catching_errors
+    } on ArgumentError catch (__) {
+      // See https://github.com/boyan01/system_clock/issues/5
+      return DateTime.now().millisecondsSinceEpoch ~/ 1000;
```

Comment:
`system_clock` builds a Flutter plugin where it exposes its own C-function that internally calls `clock_gettime()`. We could cut the hoops and directly call glibc, leaving out the unnecessary extra Flutter plugin: https://github.com/canonical/ubuntu-desktop-installer/pull/591 (it uses `sysinfo::uptime` instead). With this, we won't have problems with tests.

Reply:
Ok, I went ahead and merged it: https://github.com/canonical/ubuntu-desktop-installer/pull/591